p2p/peer_tracker.go

```diff
@@ -312,10 +324,5 @@ func (p *peerTracker) blockPeer(pID libpeer.ID, reason error) {
 	}
 
 	log.Warnw("header/p2p: blocked peer", "pID", pID, "reason", reason)
-
-	p.peerLk.Lock()
-	defer p.peerLk.Unlock()
-	// remove peer from cache.
-	delete(p.trackedPeers, pID)
-	delete(p.disconnectedPeers, pID)
```

Comment:
Blocked peers will get caught by disconnected loop, that will take care of deleting them from trackedPeers but will still be kept in disconnectedPeers map - do we want this?

Reply:
The motivation for this fairly irrelevant change is not to do something we don't need to. It's anyway gets removed on the disconnect event triggered by closed connection above as well as disconnected will be cleanedup by gc